### PR TITLE
tpm: Get rid of AbstractTPM class

### PR DESCRIPTION
**Pull request by stefanberger**
AbstractTPM may have served as an interface-defining class while TPM 1.2 support was still there but now it's obsolete, so remove it.

### CODE REVIEW

File: keylime/tpm/tpm_main.py [hunk 1]
```diff
@@ -29,8 +29,9 @@
 logger = keylime_logging.init_logging("tpm")
 
 
-class tpm(tpm_abstract.AbstractTPM):
+class tpm:
```
Comment (aplanas) [inline keylime/tpm/tpm_main.py:32]: Maybe a chance to rename it as "Tpm". Usually lowercase names for classes are for primitive types (list, dict) but not for user defined classes

File: keylime/tpm/tpm_main.py [hunk 2]
```diff
@@ -29,8 +29,9 @@
...
     tools_version: str = ""
+    EXIT_SUCCESS: int = 0
```
Comment (aplanas) [inline keylime/tpm/tpm_main.py:34]: As a constant, I would put it outside the class scope

File: keylime/tpm/tpm_main.py [hunk 3]
```diff
@@ -90,7 +91,7 @@ def run(self, cmd: Sequence[str]) -> cmd_exec.RetDictType:
     def __run(
         self,
         cmd: Sequence[str],
-        expectedcode: int = tpm_abstract.AbstractTPM.EXIT_SUCESS,
+        expectedcode: int = EXIT_SUCCESS,
```
Comment (aplanas) [inline keylime/tpm/tpm_main.py:94]: that should fail, shouldn't?

**Review by THS-on — COMMENTED** [review]
Once the style and linter checks are fixed this LGTM. 

Note that https://github.com/keylime/keylime/pull/1369 also modifies the TPM code so it defintetly requires then a rebase, but I think merging this first makes sense

File: keylime/tpm/tpm_main.py [hunk 4]
```diff
@@ -399,6 +629,18 @@ def check_quote_timing(
...
 
+    @staticmethod
+    def hashdigest(payload: bytes, algorithm: Hash) -> Optional[str]:
+        digest = algorithm.hash(payload)
+        if digest is None:
+            return None
+        return codecs.encode(digest, "hex").decode("utf-8")
+
+    @staticmethod
+    def START_HASH(algorithm: Hash) -> str:
```
[the remark this answers is not in the document]
Comment (stefanberger) [reply keylime/tpm/tpm_main.py]: Fixed.

File: keylime/tpm/tpm_main.py (hunk 2 above)
Comment (stefanberger) [reply keylime/tpm/tpm_main.py]: Done.

**Comment by THS-on** [comment]
I checked the CodeQL warnings and they are false positives, because they assume agent_id can leak the mtls certificate.

**Comment by stefanberger** [comment]
> Once the style and linter checks are fixed this LGTM.

Fixed.

**Review by THS-on — APPROVED** [review]

**Pull request merged**